main.c

```diff
@@ -46,6 +46,48 @@ bool handleWriteFault(struct page_table *pt, int page) {
     return readOnly;
 }
 
+void custom_handler(struct page_table *pt, int page ) {
+    int frame = -1;
+    num_faults++;
+
+    if(handleWriteFault(pt, page)) return;
+
+    for(int f=0;f<pt->nframes;f++){
+
+        // Find a frame that is not already allocated to a page.
+        bool success = true;
+        for (int curr_page=0; curr_page < pt->npages; curr_page++){
+            if(pt->page_mapping[curr_page] == f && pt->page_bits[curr_page] != 0){
+                success = false;
+                break;
+            }
+        } if(success){
+            frame = f;
+            break;
+        }
+    }
+
+    // Find the page that is the greatest distance mod npages from the new page
+    if(frame == -1){
+	int page_out = 0;
+	int dist;
+	int max_dist = 0;
+	for (int curr_page=0; curr_page < pt->npages; curr_page++){
+	    if(pt->page_bits[curr_page] != 0) {
+		dist = (page - curr_page) % pt->npages;
```

Comment:
Logically I think we should handle the case when dist is negative here

Reply:
Actually @kstowe , this is actually evicting the last page that is attached to a frame, rather than the frame that is most distance. This is because we aren't checking when the distance is negative, so we are effectively picking the last page (furthest from the current and ahead). This does mean we have a great page fault algorithm tho.